packages/lit-element/src/lib/updating-element.ts

```diff
@@ -640,27 +535,28 @@ export abstract class UpdatingElement extends HTMLElement {
    */
   attributeChangedCallback(
     name: string,
-    old: string | null,
+    _old: string | null,
     value: string | null
   ) {
-    if (old !== value) {
-      this._attributeToProperty(name, value);
-    }
+    this._attributeToProperty(name, value);
   }
 
   private _propertyToAttribute(
     name: PropertyKey,
     value: unknown,
     options: PropertyDeclaration = defaultPropertyDeclaration
   ) {
-    const ctor = this.constructor as typeof UpdatingElement;
-    const attr = ctor._attributeNameForProperty(name, options);
-    if (attr !== undefined) {
-      const attrValue = ctor._propertyValueToAttribute(value, options);
-      // an undefined value does not change the attribute.
-      if (attrValue === undefined) {
-        return;
-      }
+    const attr = (this
+      .constructor as typeof UpdatingElement).attributeNameForProperty(
+      name,
+      options
+    );
+    if (attr !== undefined && options.reflect) {
+      const converter = options.converter;
+      const toAttribute =
+        (converter && (converter as ComplexAttributeConverter).toAttribute) ||
+        defaultConverter.toAttribute;
```

Comment:
Use optional chaining and nullish coalescing here:

```suggestion
      const toAttribute =
        ((options.converter as ComplexAttributeConverter)?.toAttribute) ??
        defaultConverter.toAttribute;
```

Reply:
Done.